crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,1008 @@
+pub struct SyncResponseBankRedirect {
+    #[serde(rename = "PaymentInformation")]
+    pub payment_information_response: BankPaymentInformationResponse,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum TrustpayPaymentsResponse {
+    CardsPayments(PaymentsResponseCards),
+    BankRedirectPayments(PaymentsResponseBankRedirect),
+    BankRedirectSync(SyncResponseBankRedirect),
+    BankRedirectError(ErrorResponseBankRedirect),
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, TrustpayPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            TrustpayPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let (status, error, payment_response_data) =
+            get_trustpay_response(item.response, item.http_code)?;
+        Ok(Self {
+            status,
+            response: error.map_or_else(|| Ok(payment_response_data), Err),
+            ..item.data
+        })
+    }
+}
+
+fn handle_cards_response(
+    response: PaymentsResponseCards,
+    status_code: u16,
+) -> CustomResult<
+    (
+        enums::AttemptStatus,
+        Option<types::ErrorResponse>,
+        types::PaymentsResponseData,
+    ),
+    errors::ConnectorError,
+> {
+    let (status, msg) = get_transaction_status(
+        response.payment_status.as_str(),
+        response.redirect_url.clone(),
+    )?;
+    let form_fields = response
+        .redirect_params
+        .unwrap_or(std::collections::HashMap::new());
+    let redirection_data = response.redirect_url.map(|url| services::RedirectForm {
+        endpoint: url,
+        method: services::Method::Post,
+        form_fields,
+    });
+    let error = if msg.is_some() {
+        Some(types::ErrorResponse {
+            code: response.payment_status,
+            message: msg.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+            reason: None,
+            status_code,
+        })
+    } else {
+        None
+    };
+    let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+        resource_id: types::ResponseId::ConnectorTransactionId(response.instance_id),
+        redirection_data,
+        mandate_reference: None,
+        connector_metadata: None,
+    };
+    Ok((status, error, payment_response_data))
+}
+
+fn handle_bank_redirects_response(
+    response: PaymentsResponseBankRedirect,
+) -> CustomResult<
+    (
+        enums::AttemptStatus,
+        Option<types::ErrorResponse>,
+        types::PaymentsResponseData,
+    ),
+    errors::ConnectorError,
+> {
+    let status = enums::AttemptStatus::AuthenticationPending;
+    let error = None;
+    let form_fields: HashMap<String, String> = HashMap::from_iter(
```

Comment:
use the `RedirectData::from` function to achieve this.

Reply:
Done